server/odc-service/src/main/java/com/oceanbase/odc/service/monitor/MeterHolder.java

```diff
@@ -0,0 +1,363 @@
+                    return;
+                }
+                // ensure lock same obj
+                this.METER_MAP.computeIfAbsent(meterKey,
+                        k -> new TimerHolder.TimerSampleHolder(null));
+            }
+            MeterKey lockKey = getLockKey(meterKey);
+            synchronized (lockKey) {
+                TimerHolder.TimerSampleHolder placeholder = this.METER_MAP.get(meterKey);
+                if (placeholder.sample != null) {
+                    return;
+                }
+                Timer.Sample start = Timer.start(this.registry);
+                this.METER_MAP.put(lockKey,
+                        new TimerHolder.TimerSampleHolder(start));
+            }
+        }
+
+        public void record(MeterKey meterKey) {
+            TimerHolder.TimerSampleHolder timerSampleHolder = this.METER_MAP.get(meterKey);
+            if (timerSampleHolder == null) {
+                return;
+            }
+            synchronized (timerSampleHolder) {
+                timerSampleHolder = this.METER_MAP.get(meterKey);
+                if (timerSampleHolder == null) {
+                    return;
+                }
+                Iterable<Tag> tags = meterKey.getTags();
+                Timer.Builder builder = Timer.builder(meterKey.getMeterName().getMeterName())
+                        .description(meterKey.getMeterName().getDescription());
+                for (Tag tag : tags) {
+                    builder.tag(tag.getKey(), tag.getValue());
+                }
+                timerSampleHolder.stop(builder.register(registry));
+            }
+        }
+
+
+        private MeterKey getLockKey(MeterKey meterKey) {
+            return this.METER_MAP.keySet().stream().filter(meterKey::equals).findFirst()
+                    .orElseThrow(IllegalAccessError::new);
+        }
+
+        @Override
+        public synchronized void clearAfterPull() {
+            this.METER_MAP.entrySet().stream().filter(e -> e.getKey().needRemove)
+                    .filter(e -> e.getValue().getMarkToRemove())
+                    .forEach(e -> {
+                        this.METER_MAP.remove(e.getKey());
+                        registry.remove(e.getValue().getTimer());
+                    });
+        }
+
+        public Integer getSize() {
+            return METER_MAP.size();
+        }
+
+        @Getter
+        public static class TimerSampleHolder {
+            Timer.Sample sample;
+            Timer timer;
+            Boolean markToRemove = false;
+
+            protected TimerSampleHolder(Timer.Sample sample) {
+                this.sample = sample;
+            }
+
+            protected synchronized void stop(Timer timer) {
+                this.timer = timer;
+                this.sample.stop(timer);
+                this.markToRemove = true;
+            }
+        }
+    }
+
+    @Value
+    public static class MeterKey {
+        MeterName meterName;
+        Iterable<Tag> tags;
+        String uniqueKey;
```

Comment:
meter Name with tags can't be global unique?
what unique key declared for
need remove also should be MeterKey fields
Generally tag is not easy recorded, Suggest make unique symbols easy to construct and record.

Reply:
`Meter Name` and` tag` are unique key of meter registery, however, not unique in manager, cause same meter may have multi sample, we have to distinguish a specific sample from other.

`Tag` is declare by `micrometer`.  add a Builer to easier to construct.